Assignment2_q1.py

```diff
@@ -0,0 +1,84 @@
+# coding=utf-8
+"""Prints all the ancestors of the key in the given binary tree. Uses two methods:
+   first method 'print_ancestors' takes a tree and a key and calls the second helper
+   recursive method 'get_ancestors_recursive' which also takes tree and a key but it
+   also passes a list of ancestors so far as an argument.
+   Given solution takes θ(n) time, where n is the number of nodes in the tree."""
+
+
+from BinaryTree import TreeNode
+
+
+def print_ancestors(tree, key):
+    """Prints all the ancestors of the key in the given binary tree and returns a list of them.
+       Ancestors contain the key itself. Ancestors are ordered: first in the list (and first printed)
+       is the key itself then it 'crawls up' the tree up until the root.
+
+        Args:
+            tree: a binary tree, tree to check for ancestors, no duplicated keys allowed
+            key: a key type in a given tree, a key for the ancestors to be printed
+
+        Returns:
+            a list of key type, list of the ancestors of the key and prints ancestors
+
+        Raises:
+            ValueError: if the given tree is empty
+            KeyError: if the given tree does not contain the key
+    """
+    # Raise an error if the tree is empty
+    if tree is None:
+        raise ValueError("Empty tree!")
+
+    # Call the helper recursive method and pass an empty list of ancestors as last argument.
```

Comment:
Not sure I understand the comment about the last argument (the last argument here is `key`).